src/Stratis.Bitcoin.Features.Consensus/Rules/ConsensusRules.cs

```diff
@@ -181,4 +187,36 @@ public void SetErrorHandler(ConsensusRule errorHandler)
             }
         }
     }
+
+    public static class ConsensusRulesExtension
+    {
+        /// <summary>
+        /// Try to find a consensus rule in the rules collection.
+        /// </summary>
+        /// <typeparam name="T">The type of rule to find.</typeparam>
+        /// <param name="rules">The rules to look in.</param>
+        /// <returns>The rule or <c>null</c> if not found in the list.</returns>
+        public static T TryFindRule<T>(this IEnumerable<ConsensusRule> rules) where T : ConsensusRule
+        {
+            return rules.OfType<T>().FirstOrDefault();
+        }
+
+        /// <summary>
+        /// Find a consensus rule in the rules collection or throw an exception of not found.
+        /// </summary>
+        /// <typeparam name="T">The type of rule to find.</typeparam>
+        /// <param name="rules">The rules to look in.</param>
+        /// <returns>The rule or <c>null</c> if not found in the list.</returns>
+        public static T FindRule<T>(this IEnumerable<ConsensusRule> rules) where T : ConsensusRule
```

Comment:
Why throw an exception? I'd return null.

Reply:
The is the complementary method to `TryXXX` this will throw if not found.